model-optimizer/extensions/middle/prelu.py

```diff
@@ -0,0 +1,66 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from extensions.ops.transpose import Transpose
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, rename_nodes
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.broadcast import Broadcast
+from mo.utils.error import Error
+
+
+class PReLUMiddleReplacer(MiddleReplacementPattern):
+    enabled = True
+
+    def run_before(self):
+        from extensions.middle.pass_separator import MiddleFinish
+        return [MiddleFinish]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('prelu', dict(op='PReLU')),
+            ],
+            edges=[
+            ])
+
+    def replace_pattern(self, graph: Graph, match: dict):
+        prelu = match['prelu']
+        input_shape = prelu.in_port(0).data.get_shape()
+        # PRelu needs to be executed in NCHW mode for input tensor of any rank due to PRelu Depthwise implementation
```

Comment:
The `PRelu Depthwise implementation` is plugin specific so we cannot write such a comment here.

Reply:
Fixed.